app/models/user/external_user.rb

```diff
@@ -0,0 +1,44 @@
+# Specialisations for handling requests made externally and mirrored to
+# Alaveteli.
+class User::ExternalUser < User
+  def initialize(info_request:)
+    @info_request = info_request
+  end
+
+  def name
+    info_request.external_user_name || _('Anonymous user')
+  end
+
+  def url_name
+    fake_slug = MySociety::Format.simplify_url_part(name, 'external_user', 32)
+    (info_request&.public_body&.url_name || '') + '_' + fake_slug
```

Comment:
Looks like there's an issue with this actually

Reply:
Production request:

```ruby
r = InfoRequest.find 187945
r.external_user_name
# => nil
# r.user_name_slug
# => "brighton_and_hove_city_council_anonymous"
```

Dev request:

```
r = InfoRequest.find 115
r.external_user_name
# => nil
r.user_name_slug
# => "tgq_anonymous_user"
```

The problem is that we always return a `name` now, whereas before we were doing:

```
      if external_user_name.nil?
        fake_slug = "anonymous"
      else
        fake_slug = MySociety::Format.simplify_url_part(external_user_name, 'external_user', 32)
      end
```

Reply:
That said, I'm not sure this even worked previously: https://whatdotheyknow.com/user/brighton_and_hove_city_council_anonymous => 404

Reply:
If we don't think we're going to break URLs we could just hard code this.

```ruby
def url_name
  'anonymous'
end
```

Reply:
omg its all over the place:

* `app/views/admin_request/show.html.erb:86`
* `LinkToHelper#user_link_for_request`
* `LinkToHelper#user_admin_link_for_request`
* `LinkToHelper#request_user_link_absolute`
* `LinkToHelper#request_user_link`